src/HttpRequest.cc

```diff
@@ -738,3 +738,31 @@ HttpRequest::manager(const CbcPointer<ConnStateData> &aMgr, const AccessLogEntry
     }
 }
 
+const Ip::Address *
+FindListeningPortAddress(const HttpRequest *request, const bool derive_ips, const Ip::Address *cacheip, const Ip::Address *localip)
+{
+    if (request) {
+        if (request->flags.interceptTproxy || request->flags.intercepted) {
+            if (!cacheip && derive_ips && request->masterXaction->squidPort)
+                cacheip = &(request->masterXaction->squidPort->s);
+            if (cacheip)
```

Comment:
For intercepted requests, the logic looks rather inconsistent: 

* On one hand, we restrict intercepted requests to cacheip if cacheip isAnyAddr().
* On the other hand, we completely ignore the fact that the request was intercepted and look at localip if cacheip is nil.

It feels like we should never (or always) look at localip when cacheip (derived or original) did not yield the information we wanted for an intercepted request.

Reply:
Once again, just using same logic as original code. Even I had those queries in my mind wrt cacheIp / localIp. But I didnt want to alter any logic without knowing how it all works internally.